index.js

```diff
@@ -67,6 +87,27 @@ module.exports = function (src, dest, opts) {
 				resolve(true);
 			});
 
+			read.on('data', function () {
+				var written = write.bytesWritten;
+				var percent = (written / progress.stat.size).toPrecision(2);
+				opts.onProgress({
+					src: src,
+					dest: dest,
```

Comment:
Events would be easier to consume. It is common practice to provide enough information, so that the consumer doesn't need to determine the associated observable - often the observable itself is provided, but we don't have one (e.g. a copy task object).

E.g. to display progress per file the consumer have to create event handlers for every single task to transform every single event to inject `dest` and/or `src`:
```
a.txt [##################  ] 90%
b.txt [################### ] 95%
```

```js
// instead of:
Promise.all(tasks.map(task => {
  const copy = cpFile(task.src, task.dest);
  copy.on('progress', progressHandler);

  return copy;
}));

// create for every task an event handler
Promise.all(tasks.map(task => {
  const copy = cpFile(task.src, task.dest);
  copy.on('progress', function (progress) {
    // transform every event
    progress.src = task.src;
    progress.dest = task.dest;
    progressHandler(progress);
  });

  return copy;
}));
```